test/integration/setup/adminclient/client.go

```diff
@@ -0,0 +1,305 @@
+	}
+
+	_, err := client.BatchCreateFederatedBundle(ctx, &bundle.BatchCreateFederatedBundleRequest{
+		Bundle: []*types.Bundle{
+			{
+				TrustDomain: "foo",
+				X509Authorities: []*types.X509Certificate{
+					{
+						Asn1: blk.Bytes,
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		log.Fatalf("Fails to create federated bundle: %v", err)
+	}
+
+	_, err = client.BatchUpdateFederatedBundle(ctx, &bundle.BatchUpdateFederatedBundleRequest{
+		Bundle: []*types.Bundle{
+			{
+				TrustDomain: "foo",
+			},
+		},
+	})
+	if err != nil {
+		log.Fatalf("Fails to update federated bundle: %v", err)
+	}
+
+	_, err = client.BatchSetFederatedBundle(ctx, &bundle.BatchSetFederatedBundleRequest{
+		Bundle: []*types.Bundle{
+			{
+				TrustDomain: "bar",
+				X509Authorities: []*types.X509Certificate{
+					{
+						Asn1: blk.Bytes,
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		log.Fatalf("Fails to set federated bundle: %v", err)
+	}
+
+	_, err = client.ListFederatedBundles(ctx, &bundle.ListFederatedBundlesRequest{})
+	if err != nil {
+		log.Fatalf("Fails to list federated bundle: %v", err)
+	}
+
+	_, err = client.GetFederatedBundle(ctx, &bundle.GetFederatedBundleRequest{
+		TrustDomain: "foo",
+	})
+	if err != nil {
+		log.Fatalf("Fails to get federated bundle: %v", err)
+	}
+
+	_, err = client.BatchDeleteFederatedBundle(ctx, &bundle.BatchDeleteFederatedBundleRequest{
+		TrustDomains: []string{"foo"},
+	})
+	if err != nil {
+		log.Fatalf("Fails to delete federated bundle: %v", err)
+	}
+}
+
+// entryRPCs verify access to Entry RPCs
+func entryRPCs(ctx context.Context, conn *grpc.ClientConn, td spiffeid.TrustDomain) {
+	log.Println("Verifing Bundle RPCs")
+	client := entry.NewEntryClient(conn)
+
+	testEntry := &types.Entry{
+		ParentId: &types.SPIFFEID{
+			TrustDomain: td.String(),
+			Path:        "foo",
+		},
+		SpiffeId: &types.SPIFFEID{
+			TrustDomain: td.String(),
+			Path:        "bar",
+		},
+		Selectors: []*types.Selector{
+			{
+				Type:  "unix",
+				Value: "uid:1001",
+			},
+		},
+	}
+
+	resp, err := client.BatchCreateEntry(ctx, &entry.BatchCreateEntryRequest{
+		Entries: []*types.Entry{
+			testEntry,
+		},
+	})
+	if err != nil {
+		log.Fatalf("Fails to create entry: %v", err)
+	}
+
+	newEntry := resp.Results[0].Entry
+
+	_, err = client.ListEntries(ctx, &entry.ListEntriesRequest{})
+	if err != nil {
+		log.Fatalf("Fails to list entries: %v", err)
+	}
+
+	_, err = client.GetEntry(ctx, &entry.GetEntryRequest{
+		Id: newEntry.Id,
+	})
+	if err != nil {
+		log.Fatalf("Fails to get entry: %v", err)
+	}
+
+	_, err = client.BatchUpdateEntry(ctx, &entry.BatchUpdateEntryRequest{
+		Entries: []*types.Entry{
+			newEntry,
+		},
+	})
+	if err != nil {
+		log.Fatalf("Fails to update entry: %v", err)
+	}
+
+	_, err = client.BatchDeleteEntry(ctx, &entry.BatchDeleteEntryRequest{
+		Ids: []string{newEntry.Id},
+	})
+	if err != nil {
+		log.Fatalf("Fails to delete entry: %v", err)
+	}
+}
+
+// agentRPCs verify access to Agent RPCs
```

Comment:
Not sure if "verifying access" is completely accurate.

```suggestion
// agentRPCs verifies that Agent RPCs run successfully
```